packages/discord.js/src/structures/User.js

```diff
@@ -188,12 +204,32 @@ class User extends Base {
   }
 
   /**
-   * The Discord "tag" (e.g. `hydrabolt#0001`) for this user
+   * The tag of this user
+   * <info>This user's username, or `hydrabolt#0001` if they're using the legacy username system</info>
    * @type {?string}
    * @readonly
+   * @deprecated Use {@link User#username} instead.
    */
   get tag() {
-    return typeof this.username === 'string' ? `${this.username}#${this.discriminator}` : null;
+    if (!tagDeprecationEmitted) {
+      process.emitWarning('User#tag is deprecated. Use User#username instead.', 'DeprecationWarning');
```

Comment:
can't deprecate method be used there?

Reply:
This is a getter, I can't just assign it there

Reply:
Asi remember it was somehow possible

Reply:
you could use Object.defineProperty/getOwnPropertyDescriptor but eh kind of not nessercary